src/fprime_extras/docs/cli.py

```diff
@@ -44,7 +54,18 @@ def docs_main(args=None, parser=None):
 
 
     print('Loading file {}'.format(args.base_file))
-    xml_file = ExtrasFile(args.base_file)
+    # Ignore the rest of this file, its a hack, and probably needs
+    # to be removed since it was written to auto-code files that are
+    # actively being removed from F Prime.
+    print(ai_file_type)
+    if ai_file_type == "Component":
+        generate_component_documentation(args.base_file)
+    elif ai_file_type == "TopologyApp":
+        generate_topology_documentation(args.base_file, args.fprime_root)
+
```

Comment:
Since these got refactored into methods/functions, lets go one step further and put them into an adjacent python file in package: `generate.py`.  That will keep this file from getting huge and seeing a lot of churn.

Reply:
Do you think the other methods should be put into a parse.py file as well or are they fine where they are? I think this could avoid a circular dependency between cli.py and generate.py. I think that would still work, but it seems like a confusing thing that should be avoided.

Reply:
I decided to go with the parse.py file. I think it works well and can probably be reused more than the generate.py file in other places

Reply:
ok, great, you beat me to it.  It is ok for things to get moved around at this stage, but just do you know: there is ``model`` package (in the works) that will be used house the general code that ingests ``Ai.xml`` files.  The models will be more complete than that, building all the connections and representing different aspects components and topology as a whole.

Reply:
Cool! Of course there are also the fprime files for parsing the Ai.xml files that we could use to, but that would require us to integrate into fprime, or at least make it a dependancy

Reply:
yea, so, I actually have a slightly different approach than the core F Prime parsers, and I want to avoid the dependency on the cmake targets.  So, we won't be able to do that (but it would be cool if they decided that our parsers were better and folded them into the upstream project).

Reply:
For clarity: I am trying to wholistic approach to model generation and code management, whereas the upstream parsers and models are designed specifically for autocoding and implementation generation.  It turns out, there are some very different requirements for those purposes (and I am not trying to replace the autocoder during the build process).

Reply:
This all seems good to go now.